odata-to-abstract-sql.js

```diff
@@ -429,6 +429,12 @@
                 }, function() {
                     return resource.name;
                 });
+                this._opt(function() {
+                    this._pred(relationshipMapping.length > 1 && relationshipMapping[0] === resource.idField);
+                    return function() {
+                        throw new Error('Attempted to directly fetch a virtual field: "' + resourceField + '"');
```

Comment:
Will this error be sent to the user? It feels very unhelpful in that case, we don't have any documentation on virtual fields. I'm not sure if there is anything we can do better here, but it's worth thinking about.

Reply:
I believe it will, and I do agree it's not a great message (I actually didn't even know it was possible to attempt this until the bug was reported to me!) but it does specify the field name you tried to fetch that you weren't allowed to which should give enough information that people can fix the problem.  Apart from that we should never be including virtual fields in the whitelist and so it would get stripped in advance, so it mainly helps internal usage.

I am 100% happy to change the message if you can think of something more useful, just as long as the field name remains part of the message

Reply:
Yeah that's the main problem for me too, I don't really know what would be better to replace it with. I didn't realise the whitelist would strip these so it should be a super rare incident that user's specifically see it, and as you say, it will show the field name that's incorrect. Cool, happy to approve.